From f7746caa935ca44708684f4bd5b8cedbe7224961 Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Tue, 19 Aug 2014 22:03:03 -0400
Subject: [crypto] pkcs7: X.509 certificate issuer and subject are mandatory fields in the ASN.1

Message-id: <1408485785-777-32-git-send-email-vgoyal@redhat.com>
Patchwork-id: 88385
O-Subject: [RHEL7.1 PATCH 31/33] PKCS#7: X.509 certificate issuer and subject are mandatory fields in the ASN.1
Bugzilla: 1129779
RH-Acked-by: David Howells <dhowells@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

BZ:1129779

commit 412eccbadfbb1521e62ae53db57d782d6bc36993
Author: David Howells <dhowells@redhat.com>
Date: Thu, 31 Jul 2014 14:46:44 +0100

X.509 certificate issuer and subject fields are mandatory fields in the ASN.1
and so their existence needn't be tested for.  They are guaranteed to end up
with an empty string if the name material has nothing we can use (see
x509_fabricate_name()).

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: David Howells <dhowells@redhat.com>
Acked-by: Vivek Goyal <vgoyal@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/crypto/asymmetric_keys/pkcs7_verify.c b/crypto/asymmetric_keys/pkcs7_verify.c
index 51ff36f..c62cf80 100644
--- a/crypto/asymmetric_keys/pkcs7_verify.c
+++ b/crypto/asymmetric_keys/pkcs7_verify.c
@@ -190,14 +190,12 @@ static int pkcs7_verify_sig_chain(struct pkcs7_message *pkcs7,
   if (ret < 0)
    return ret;
 
-  if (x509->issuer)
-   pr_debug("- issuer %s\n", x509->issuer);
+  pr_debug("- issuer %s\n", x509->issuer);
   if (x509->authority)
    pr_debug("- authkeyid %s\n", x509->authority);
 
   if (!x509->authority ||
-      (x509->subject &&
-       strcmp(x509->subject, x509->issuer) == 0)) {
+      strcmp(x509->subject, x509->issuer) == 0) {
    /* If there's no authority certificate specified, then
     * the certificate must be self-signed and is the root
     * of the chain.  Likewise if the cert is its own
-- 
1.7.1